src/spikeinterface/core/base.py

```diff
@@ -920,16 +935,16 @@ def save_to_zarr(
         return cached
 
 
-def _make_paths_relative(d, relative) -> dict:
-    relative = str(Path(relative).absolute())
-    func = lambda p: os.path.relpath(str(p), start=relative)
-    return recursive_path_modifier(d, func, target="path", copy=True)
+def _make_paths_relative(d, relative, copy=True) -> dict:
+    relative = Path(relative).absolute()
+    func = lambda p: os.path.relpath(p, start=relative)
```

Comment:
`os.path` is compatible with `pathlib` objects

Reply:
There is an equivalent pathlib function as well:
https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.relative_to

Reply:
It doesn't work if paths are not direct relatives (i.e. it doesn't find .. locations)

Reply:
I can attest to that: I had a similar problem a few months back, and the `pathlib` equivalent did not work, and I had to use `os.path.relpath`. (link to where I used it: https://github.com/BarbourLab/lussac/blob/9c71ed6ad1303460566bf46bacf22c25ee805fe3/src/lussac/utils/plotting.py#L26)

Reply:
Thanks.
This is good to know. It is on a footnote on the link the I shared actually:

```
[PurePath.relative_to()](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.relative_to) requires self to be the subpath of the argument, but [os.path.relpath()](https://docs.python.org/3/library/os.path.html#os.path.relpath) does not.
```

Double thanks, guys. I learn a couple of things about paths today. This is an enhancement that will be added in python 3:12, see the argument `walk_up` in the docs over there:

https://docs.python.org/3.12/library/pathlib.html#pathlib.PurePath.relative_to

The corresponding PR:
https://github.com/python/cpython/pull/19813